packages/bridge/src/vl/utils/conversion/category/index.ts

```diff
@@ -10,11 +10,16 @@ import { findColorForCategory } from '../histogram';
  * @returns {object[]}
  */
 export function vlToCategory(histogram: VLCategoricalHistogram, legendData: any = []): object[] {
-  return histogram.value.map((d) => ({
-    color: findColorForCategory(d.x, legendData),
-    name: d.x,
-    value: d.y
-  }));
+  const fullCategories = (histogram as any)._categories;
+  const values = histogram.value;
+  return fullCategories.map((d) => {
+    const value = values.find((elem) => elem.x === d.name);
+    return {
+      color: findColorForCategory(d.name, legendData),
+      name: d.name,
+      value: value ? value.y : 0
```

Comment:
I agree with @Jesus89 that 0 is not the proper value for a category that doesn't exist within the current viewport. We can make some tweaks to the category widget and avoid printing the value if it is undefined or something like that.

Reply:
Sure, but will it change for everybody or will we set a flag/option to define this behavior in order to avoid breaking things to current Airship users in a minor version?

Reply:
I don't know, my take would be to do the change for all users. It's not a major change IMHO.

If value is undefined, no value is shown.
If value is null, 0 is shown.
If value is a number, the number is shown then.

But let's see what other participants in the PR think.

Reply:
If undefined does the trick, let's go for it. IMO

Reply:
Didn't try with `undefined` yet, but anyway, if any user currently uses `undefined` and expects a 0 it will not be like this anymore, so this tricky behavior needs at least be documented, though I wouldn't recommend this `null` vs `undefined` trick unless it is more or less standard.
Anyway, I will try it.

Reply:
I would go for it too

Reply:
`Undefined` doesn't work, it draws `NaN` in the widget and somehow breaks sorting items. Not bad as long as nobody will be using this and it should be ease to fix, though I still don't recommend this trick, I rather treat `undefined` and `null` values the same or add a `hideNulls` like flag.
Anyway, both options should work, so which do you prefer?

Reply:
I completely agree on this. I would not use undefined and null with those different meanings

IMO:

If value is null, no value is shown.
If value is a number, the number is shown then (also if we have a true 0, of course)

If value is undefined,.... Do we ever get undefined? Why?

Reply:
Agree. I think we tried undefined instead of null to check if it does the trick, but if it doesn't work we should use a defined null to set "empty" value and then fix the widget to support that